contrib/shibuqam/examples/shibuqamoauth.nit

```diff
@@ -0,0 +1,157 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Server that implements an authentication bound to a `shibuqam` valid server.
+#
+# # Use as a web service
+#
+# ## Request
+#
+# Two GET fields
+#
+# * `redir`: A full URI that will be used to redirect the user once the authentication is done.
+# * `state`: A temporary string used to check that the callback is legitimate.
+#
+# On request, the service authenticate the user then redirect to the callback with the user informations.
+#
+# ## Response
+#
+# After the authentication, the response is a redirection to `redir` with the following GET arguments:
```

Comment:
It's fine to put sensitive information in a URL. Stripe does it for legacy platforms that does not support CORS on Ajax calls. Basically instead of making ajax calls, it will just add a `<script />` element in the DOM which will make a get call. So for instance if you pay by credit card using Stripe on a browser that does not support CORS (hello IE we meet again) it'll do something like:

```
<script type="text/yajascript" src="https://api.stripe.com/v2/pay?credit_card[name]=Toto&credit_card[number]=4242&credit_card[cvv]=321"></script>
```

As you can see, it'll make a GET request with credit card numbers. Sure the browser won't save that URL in its history, but it's clearly a GET request with sensitive information. And yes, it's PCI compliant. It's stripe after all.